conans/client/generators/cmake_find_package.py

```diff
@@ -119,10 +119,10 @@ class CMakeFindPackageGenerator(Generator):
 
         {%- for public_dep in pkg_public_deps %}
 
-        if(NOT {{ public_dep }}_FOUND)
-            find_dependency({{ public_dep }} REQUIRED)
+        if(NOT {{ public_dep["name"] }}_FOUND)
+            find_dependency({{ public_dep["filename"] }} REQUIRED)
```

Comment:
I believe you cannot use this syntax here and you will have feed the template with the correct names in the `pkg_public_deps` list